crates/sovereign-sdk/rollup-interface/src/mmr/mod.rs

```diff
@@ -0,0 +1,78 @@
+#![allow(missing_docs)]
+
+use alloc::vec::Vec;
+
+use anyhow::Result;
+use borsh::{BorshDeserialize, BorshSerialize};
+use serde::{Deserialize, Serialize};
+use sha2::{Digest, Sha256};
+
+mod guest;
+#[cfg(any(feature = "native", feature = "testing"))]
+mod native;
+#[cfg(test)]
+mod tests;
+
+pub use guest::*;
+#[cfg(any(feature = "native", feature = "testing"))]
+pub use native::*;
+
+pub type MMRNodeHash = [u8; 32];
+pub type Wtxid = [u8; 32];
+
+pub trait NodeStore {
+    fn save_node(&mut self, level: usize, index: usize, hash: MMRNodeHash) -> Result<()>;
+    fn load_node(&self, level: usize, index: usize) -> Result<Option<MMRNodeHash>>;
+    fn save_chunk(&mut self, wtxid: Wtxid, chunk: MMRChunk) -> Result<()>;
+    fn load_chunk(&self, wtxid: Wtxid) -> Result<Option<MMRChunk>>;
+    fn get_tree_size(&self) -> usize;
+    fn set_tree_size(&mut self, size: usize) -> Result<()>;
+}
+
+#[derive(Serialize, Deserialize, Eq, PartialEq, Clone, Debug, BorshDeserialize, BorshSerialize)]
+pub struct MMRInclusionProof {
+    pub subroot_idx: usize,
```

Comment:
I think this could be breaking since MMRInclusionProof is part of the circuit input. usize is 64-bits serialized in native and 32-bits serialized in zk. Didn't confirm this for sure, but their spec https://borsh.io does not specify usize, which is a sign that they are using what compiler returns as the size of this int, which is platform dependent.

This also made me notice that we are using usize for some other fields as well and needs to be fixed asap.

Reply:
Thank you for sharing this. I noticed some tests failing because of an overflow problem. I'll check if this is related.